src/components/Home/HomepageMap.tsx

```diff
@@ -21,46 +23,55 @@ export default function HomepageMap() {
 
     // Create the state for the modal info
     const [activeStation, setActiveStation] = useState(0);
-    const providerData = useContext(StationsContext);
-    const handleModal = (value:boolean, stationId: number) => {
-        providerData.handleModal(value);
+    const [markers, setMarkers] = useState<any>([]);
```

Comment:
If I got it right, the markers should reflect any changes to the stations. This means that we want the component to re-render when it picks any such change. However, if you set/assign a variable inside the hook, it will be lost after each render. We need to preserve the value of the markers though. To allow this I added the useState to keep track of it in this component and re-set it inside the hook.  Please correct me if I'm wrong in the way I understood the requirement here.

Reply:
An update here, I initially had used useEffect, but change it to useMemo instead. In this specific scenario both have the same behaviour for our system, but useMemo has some extra benefits that makes me believe it's the better option here. Some useful docs:
https://react.dev/reference/react/useEffect
https://react.dev/reference/react/useMemo

Reply:
Let's talk about this, sounds like an interesting topic for discussion.